queue_services/entity-filer/src/entity_filer/filing_processors/registration.py

```diff
@@ -0,0 +1,151 @@
+                f'Queue Error: Unable to affiliate business:{business.identifier} for filing:{filing.id}',
+                level='error'
+            )
+        else:
+            # flip the registration
+            # recreate the bootstrap, but point to the new business in the name
+            old_bs_affiliation = AccountService.delete_affiliation(bootstrap.account, bootstrap.identifier)
+            new_bs_affiliation = AccountService.create_affiliation(
+                account=bootstrap.account,
+                business_registration=bootstrap.identifier,
+                business_name=business.identifier,
+                corp_type_code='RTMP'
+            )
+            reaffiliate = bool(new_bs_affiliation in (HTTPStatus.OK, HTTPStatus.CREATED)
+                               and old_bs_affiliation == HTTPStatus.OK)
+
+        if rv not in (HTTPStatus.OK, HTTPStatus.CREATED) \
+                or ('deaffiliation' in locals() and deaffiliation != HTTPStatus.OK)\
+                or ('reaffiliate' in locals() and not reaffiliate):
+            raise QueueException
+    except Exception as err:  # pylint: disable=broad-except; note out any exception, but don't fail the call
+        sentry_sdk.capture_message(
+            f'Queue Error: Affiliation error for filing:{filing.id}, with err:{err}',
+            level='error'
+        )
+
+
+def process(business: Business,  # pylint: disable=too-many-branches
+            filing: Dict,
+            filing_rec: Filing,
+            filing_meta: FilingMeta):  # pylint: disable=too-many-branches
+    """Process the incoming registration filing."""
+    # Extract the filing information for registration
+    registration_filing = filing.get('filing', {}).get('registration')
+    filing_meta.registration = {}
+
+    if not registration_filing:
+        raise QueueException(f'Registration legal_filing:registration missing from {filing_rec.id}')
+    if business:
+        raise QueueException(f'Business Already Exist: Registration legal_filing:registration {filing_rec.id}')
+
+    business_info_obj = registration_filing.get('nameRequest')
+
+    # Reserve the Corp Number for this entity
+    corp_num = business_info.get_next_corp_num('FM')
+    if not corp_num:
+        raise QueueException(
+            f'registration {filing_rec.id} unable to get a business registration number.')
+
+    # Initial insert of the business record
+    business = Business()
+    business = business_info.update_business_info(corp_num, business, business_info_obj, filing_rec)
+    business.founding_date = datetime.fromisoformat(registration_filing.get('startDate'))
+    if naics := registration_filing.get('business', {}).get('naics'):
+        naics_code = naics.get('naicsCode')
+        naics_structure = NaicsStructure.find_by_code(naics_code)
+        business.naics_key = naics_structure.naics_key
+        business.naics_code = naics_code
+        business.naics_description = naics_structure.class_title
```

Comment:
As per our conversation in backlog grooming, we have a ticket for updating the BE to do a REST call to the GET NAICS code endpoint in the filer and the registration/change filing validation code.  So I think we can remove L103 and L104 for now.  Maybe add a todo mentioning we need to do a REST call to get NAICS code endpoint to get the naics key.